Tests/AppTests/Util+ext.swift

```diff
@@ -90,7 +90,23 @@ extension Snapshotting where Value == HTML, Format == NSImage {
         Snapshotting<NSView, NSImage>.image(precision: precision, size: size).pullback { node in
             let html = node.render()
             let webView = WKWebView()
-            webView.loadHTMLString(html, baseURL: baseURL)
+            
+            if let url = baseURL {
+                let htmlURL = url.appendingPathComponent("index.html")
+                
+                // Save HTML file at root of public directory
+                do {
+                    try html.write(to: htmlURL, atomically: true, encoding: .utf8)
+                } catch {
+                    print("💥 Failed to write index.html: \(error)")
+                }
+                
+                // Load the HTML file into the web view with access to public directory
+                webView.loadFileURL(htmlURL, allowingReadAccessTo: url)
```

Comment:
Ouf, having to write out the files is a bummer. I wonder if the resource location would have any impact on `loadHTMLString`.

Reply:
Unfortunately I don't believe so - I did a hell of a lot of investigations with this and a lot of StackOverflow comments pointed me in the direction of this method.

It seems to all be to do with the file structure and where the resources are relative to the HTML. WebKit (for security reasons) really doesn't like having file:// resources in a different directory - it needs to be a child of the same parent.

This is ultimately what the whole solution pivots on